controllers/feeds/getFeedsV2.js

```diff
@@ -94,6 +94,71 @@ const isValidActivity = async (item, conditions) => {
   return (item.final_score || 0) >= threshold;
 };
 
+const getActorFromLatestReaction = (latestReaction) => {
+  let postActors = [];
+  if (latestReaction?.upvotes?.length > 0) {
+    for (let j = 0; j < latestReaction?.upvotes.length; j++) {
+      if (!postActors.includes(latestReaction.upvotes[j].user_id)) {
+        postActors.push(latestReaction.upvotes[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.downvotes?.length > 0) {
+    for (let j = 0; j < latestReaction?.downvotes.length; j++) {
+      if (!postActors.includes(latestReaction.downvotes[j].user_id)) {
+        postActors.push(latestReaction.downvotes[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.comments?.length > 0) {
+    for (let j = 0; j < latestReaction?.comments.length; j++) {
+      if (!postActors.includes(latestReaction.comments[j].user_id)) {
+        postActors.push(latestReaction.comments[j].user_id);
+      }
+    }
+  }
+  if (latestReaction?.comment?.length > 0) {
+    for (let j = 0; j < latestReaction?.comment.length; j++) {
+      if (!postActors.includes(latestReaction.comment[j].user_id)) {
+        postActors.push(latestReaction.comment[j].user_id);
+      }
+    }
+  }
+  return postActors;
+};
+
+const setKarmaScore = (latestReaction, karmaScores) => {
+  return latestReaction.map((reaction) => {
+    const user = karmaScores.find((user) => user.user_id === reaction.user_id);
+    if (reaction.latest_children) {
+      reaction.latest_children = addKarmaScoreToLatestReaction(
+        reaction.latest_children,
+        karmaScores
+      );
+    }
+    return {
+      ...reaction,
+      karma_score: roundingKarmaScore(user?.karma_score || 0)
+    };
+  });
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `setKarmaScore` function, you are modifying the `reaction.latest_children` property directly. It's generally not a good practice to mutate function arguments as it can lead to unexpected side effects. Instead, create a copy of the object and then modify it.

```diff
-    if (reaction.latest_children) {
-      reaction.latest_children = addKarmaScoreToLatestReaction(
-        reaction.latest_children,
-        karmaScores
-      );
-    }
+    const updatedReaction = { ...reaction };
+    if (updatedReaction.latest_children) {
+      updatedReaction.latest_children = addKarmaScoreToLatestReaction(
+        updatedReaction.latest_children,
+        karmaScores
+      );
+    }
-    return {
-      ...reaction,
-      karma_score: roundingKarmaScore(user?.karma_score || 0)
-    };
+    updatedReaction.karma_score = roundingKarmaScore(user?.karma_score || 0);
+    return updatedReaction;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->